product_packaging_type_required/models/product_packaging_type.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 Camptocamp (<http://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class ProductPackagingType(models.Model):
+    _inherit = "product.packaging.type"
+
+    required = fields.Boolean()
+
+    @api.model
+    def cron_check_create_required_packaging(self, limit=0):
+        """ if limit=0, the method will not apply a limit to process missing
+        packages.
+        """
+        existing_products = self.env['product.product'].search(
+            [('type', 'in', ('product', 'consu'))]
+        )
+        i = 0
+        for p_type in self.search([('required', '=', True)]):
+            if limit and i == limit:
+                break
+            for product in existing_products:
+                packaging = self.env['product.packaging'].search(
+                    [('product_id', '=', product.id),
+                     ('packaging_type_id', '=', p_type.id)]
+                )
+                if packaging:
+                    continue
+                else:
+                    packaging.create(
+                        self.prepare_packaging_vals(product, p_type)
```

Comment:
```suggestion
                        p_type.prepare_packaging_vals(product)
```
This looks a bit more logical...